libtenzir/src/argument_parser2.cpp

```diff
@@ -171,6 +171,21 @@ auto argument_parser2::parse(const ast::entity& self,
           return;
         }
         auto cast = caf::get_if<T>(&*value);
+        if constexpr (std::same_as<T, uint64_t>) {
+          if (not cast) {
+            auto other = caf::get_if<int64_t>(&*value);
+            if (other) {
+              if (*other < 0) {
+                emit(diagnostic::error("expected positive integer, got `{}`",
+                                       *other)
+                       .primary(expr));
+                return;
+              }
+              value = static_cast<uint64_t>(*other);
+              cast = caf::get_if<T>(&*value);
+            }
+          }
+        }
```

Comment:
@jachris this is more or less a copy of the handling of the same situation in another place in the same file. Feel free to unify the two. I ran into this because I had a `std::optional<located<uint64_t>>` in an `argument_parser2`, which didn't convert as I had expected it to.

Reply:
Thanks :+1: 

Side note: The type promotion thing is still has a bit to be desired in general, as it also becomes relevant when you define functions etc... Doing everything manually for now until we get a feeling for what we want to do.